### DISCUSSION TITLE
[Perf] Real app startup time

### DISCUSSION THREAD

**Discussion by lenoirzamboni** [invertase/react-native-firebase#5425] [Q&A]
Hello!

I want to know if the real app startup load time will be recorded on the firebase console. If no, is there a way to do this with the `react-native-firebase/perf`?

**Comment by mikehardy** [comment]
Hi there! This is a general question better suited to stackoverflow or the underlying documentation of either firebase-ios-sdk or firebase-android-sdk depending on the platform you are interested in, as I do not believe there is anything react-native-firebase specific involved.

If I'm wrong that there is nothing react-native-firebase specific and by "real app startup time" you mean

1) "time to interactive from cold start to first screen interactive" (note: all those terms are precise, and it is important to be precise here) 
2) and for some reason the built in performance timers stop timing app startup as soon as the Android MainActivity is loaded or the iOS view is loaded, which is when the react-native bundle is loading then you may be able to start a second timer in native code just before the bundle load starts and finish the timer in react-native code at whatever point you consider your first react-native screen to be interactive

So, not sure about your first question, though you may answer it via experiment
For the second question, if you need some react-native way to time things not otherwise included, the sketch above should help
